// @getbrevo/brevo/dist/model/contactErrorModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactErrorModel = void 0;
class ContactErrorModel {
    static getAttributeTypeMap() {
        return ContactErrorModel.attributeTypeMap;
    }
}
exports.ContactErrorModel = ContactErrorModel;
ContactErrorModel.discriminator = undefined;
ContactErrorModel.attributeTypeMap = [
    {
        "name": "code",
        "baseName": "code",
        "type": "ContactErrorModel.CodeEnum"
    },
    {
        "name": "message",
        "baseName": "message",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "object"
    }
];
(function (ContactErrorModel) {
    let CodeEnum;
    (function (CodeEnum) {
        CodeEnum[CodeEnum["InvalidParameter"] = 'invalid_parameter'] = "InvalidParameter";
        CodeEnum[CodeEnum["MissingParameter"] = 'missing_parameter'] = "MissingParameter";
        CodeEnum[CodeEnum["DocumentNotFound"] = 'document_not_found'] = "DocumentNotFound";
        CodeEnum[CodeEnum["AccountInProcess"] = 'account_in_process'] = "AccountInProcess";
        CodeEnum[CodeEnum["DuplicateParameter"] = 'duplicate_parameter'] = "DuplicateParameter";
        CodeEnum[CodeEnum["MethodNotAllowed"] = 'method_not_allowed'] = "MethodNotAllowed";
        CodeEnum[CodeEnum["OutOfRange"] = 'out_of_range'] = "OutOfRange";
    })(CodeEnum = ContactErrorModel.CodeEnum || (ContactErrorModel.CodeEnum = {}));
})(ContactErrorModel || (exports.ContactErrorModel = ContactErrorModel = {}));
