// @getbrevo/brevo/dist/model/corporateUserEmailPermissionsPutRequestPrivilegesInner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CorporateUserEmailPermissionsPutRequestPrivilegesInner = void 0;
class CorporateUserEmailPermissionsPutRequestPrivilegesInner {
    static getAttributeTypeMap() {
        return CorporateUserEmailPermissionsPutRequestPrivilegesInner.attributeTypeMap;
    }
}
exports.CorporateUserEmailPermissionsPutRequestPrivilegesInner = CorporateUserEmailPermissionsPutRequestPrivilegesInner;
CorporateUserEmailPermissionsPutRequestPrivilegesInner.discriminator = undefined;
CorporateUserEmailPermissionsPutRequestPrivilegesInner.attributeTypeMap = [
    {
        "name": "feature",
        "baseName": "feature",
        "type": "CorporateUserEmailPermissionsPutRequestPrivilegesInner.FeatureEnum"
    },
    {
        "name": "permissions",
        "baseName": "permissions",
        "type": "Array<CorporateUserEmailPermissionsPutRequestPrivilegesInner.PermissionsEnum>"
    }
];
(function (CorporateUserEmailPermissionsPutRequestPrivilegesInner) {
    let FeatureEnum;
    (function (FeatureEnum) {
        FeatureEnum[FeatureEnum["UserManagement"] = 'user_management'] = "UserManagement";
        FeatureEnum[FeatureEnum["Api"] = 'api'] = "Api";
        FeatureEnum[FeatureEnum["MyPlan"] = 'my_plan'] = "MyPlan";
        FeatureEnum[FeatureEnum["AppsManagement"] = 'apps_management'] = "AppsManagement";
        FeatureEnum[FeatureEnum["Analytics"] = 'analytics'] = "Analytics";
        FeatureEnum[FeatureEnum["SubOrganizationGroups"] = 'sub_organization_groups'] = "SubOrganizationGroups";
        FeatureEnum[FeatureEnum["CreateSubOrganizations"] = 'create_sub_organizations'] = "CreateSubOrganizations";
        FeatureEnum[FeatureEnum["ManageSubOrganizations"] = 'manage_sub_organizations'] = "ManageSubOrganizations";
        FeatureEnum[FeatureEnum["Security"] = 'security'] = "Security";
    })(FeatureEnum = CorporateUserEmailPermissionsPutRequestPrivilegesInner.FeatureEnum || (CorporateUserEmailPermissionsPutRequestPrivilegesInner.FeatureEnum = {}));
    let PermissionsEnum;
    (function (PermissionsEnum) {
        PermissionsEnum[PermissionsEnum["All"] = 'all'] = "All";
        PermissionsEnum[PermissionsEnum["None"] = 'none'] = "None";
        PermissionsEnum[PermissionsEnum["Create"] = 'create'] = "Create";
        PermissionsEnum[PermissionsEnum["EditDelete"] = 'edit_delete'] = "EditDelete";
        PermissionsEnum[PermissionsEnum["CreateAlerts"] = 'create_alerts'] = "CreateAlerts";
        PermissionsEnum[PermissionsEnum["DownloadData"] = 'download_data'] = "DownloadData";
        PermissionsEnum[PermissionsEnum["MyLooks"] = 'my_looks'] = "MyLooks";
        PermissionsEnum[PermissionsEnum["ExploreCreate"] = 'explore_create'] = "ExploreCreate";
    })(PermissionsEnum = CorporateUserEmailPermissionsPutRequestPrivilegesInner.PermissionsEnum || (CorporateUserEmailPermissionsPutRequestPrivilegesInner.PermissionsEnum = {}));
})(CorporateUserEmailPermissionsPutRequestPrivilegesInner || (exports.CorporateUserEmailPermissionsPutRequestPrivilegesInner = CorporateUserEmailPermissionsPutRequestPrivilegesInner = {}));
