// @getbrevo/brevo/dist/model/createAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAttribute = void 0;
class CreateAttribute {
    static getAttributeTypeMap() {
        return CreateAttribute.attributeTypeMap;
    }
}
exports.CreateAttribute = CreateAttribute;
CreateAttribute.discriminator = undefined;
CreateAttribute.attributeTypeMap = [
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    },
    {
        "name": "isRecurring",
        "baseName": "isRecurring",
        "type": "boolean"
    },
    {
        "name": "enumeration",
        "baseName": "enumeration",
        "type": "Array<CreateAttributeEnumerationInner>"
    },
    {
        "name": "multiCategoryOptions",
        "baseName": "multiCategoryOptions",
        "type": "Array<string>"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CreateAttribute.TypeEnum"
    }
];
(function (CreateAttribute) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Text"] = 'text'] = "Text";
        TypeEnum[TypeEnum["Date"] = 'date'] = "Date";
        TypeEnum[TypeEnum["Float"] = 'float'] = "Float";
        TypeEnum[TypeEnum["Boolean"] = 'boolean'] = "Boolean";
        TypeEnum[TypeEnum["Id"] = 'id'] = "Id";
        TypeEnum[TypeEnum["Category"] = 'category'] = "Category";
        TypeEnum[TypeEnum["MultipleChoice"] = 'multiple-choice'] = "MultipleChoice";
    })(TypeEnum = CreateAttribute.TypeEnum || (CreateAttribute.TypeEnum = {}));
})(CreateAttribute || (exports.CreateAttribute = CreateAttribute = {}));
