// @getbrevo/brevo/dist/model/createTierGroupRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTierGroupRequest = void 0;
class CreateTierGroupRequest {
    constructor() {
        this['upgradeStrategy'] = CreateTierGroupRequest.UpgradeStrategyEnum.RealTime;
        this['downgradeStrategy'] = CreateTierGroupRequest.DowngradeStrategyEnum.RealTime;
    }
    static getAttributeTypeMap() {
        return CreateTierGroupRequest.attributeTypeMap;
    }
}
exports.CreateTierGroupRequest = CreateTierGroupRequest;
CreateTierGroupRequest.discriminator = undefined;
CreateTierGroupRequest.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "upgradeStrategy",
        "baseName": "upgradeStrategy",
        "type": "CreateTierGroupRequest.UpgradeStrategyEnum"
    },
    {
        "name": "downgradeStrategy",
        "baseName": "downgradeStrategy",
        "type": "CreateTierGroupRequest.DowngradeStrategyEnum"
    },
    {
        "name": "tierOrder",
        "baseName": "tierOrder",
        "type": "Array<string>"
    }
];
(function (CreateTierGroupRequest) {
    let UpgradeStrategyEnum;
    (function (UpgradeStrategyEnum) {
        UpgradeStrategyEnum[UpgradeStrategyEnum["RealTime"] = 'real_time'] = "RealTime";
        UpgradeStrategyEnum[UpgradeStrategyEnum["MembershipAnniversary"] = 'membership_anniversary'] = "MembershipAnniversary";
        UpgradeStrategyEnum[UpgradeStrategyEnum["TierAnniversary"] = 'tier_anniversary'] = "TierAnniversary";
    })(UpgradeStrategyEnum = CreateTierGroupRequest.UpgradeStrategyEnum || (CreateTierGroupRequest.UpgradeStrategyEnum = {}));
    let DowngradeStrategyEnum;
    (function (DowngradeStrategyEnum) {
        DowngradeStrategyEnum[DowngradeStrategyEnum["RealTime"] = 'real_time'] = "RealTime";
        DowngradeStrategyEnum[DowngradeStrategyEnum["MembershipAnniversary"] = 'membership_anniversary'] = "MembershipAnniversary";
        DowngradeStrategyEnum[DowngradeStrategyEnum["TierAnniversary"] = 'tier_anniversary'] = "TierAnniversary";
    })(DowngradeStrategyEnum = CreateTierGroupRequest.DowngradeStrategyEnum || (CreateTierGroupRequest.DowngradeStrategyEnum = {}));
})(CreateTierGroupRequest || (exports.CreateTierGroupRequest = CreateTierGroupRequest = {}));
