// @getbrevo/brevo/dist/model/createBalanceLimitPayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBalanceLimitPayload = void 0;
class CreateBalanceLimitPayload {
    static getAttributeTypeMap() {
        return CreateBalanceLimitPayload.attributeTypeMap;
    }
}
exports.CreateBalanceLimitPayload = CreateBalanceLimitPayload;
CreateBalanceLimitPayload.discriminator = undefined;
CreateBalanceLimitPayload.attributeTypeMap = [
    {
        "name": "constraintType",
        "baseName": "constraintType",
        "type": "CreateBalanceLimitPayload.ConstraintTypeEnum"
    },
    {
        "name": "durationUnit",
        "baseName": "durationUnit",
        "type": "CreateBalanceLimitPayload.DurationUnitEnum"
    },
    {
        "name": "durationValue",
        "baseName": "durationValue",
        "type": "number"
    },
    {
        "name": "slidingSchedule",
        "baseName": "slidingSchedule",
        "type": "boolean"
    },
    {
        "name": "transactionType",
        "baseName": "transactionType",
        "type": "CreateBalanceLimitPayload.TransactionTypeEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    }
];
(function (CreateBalanceLimitPayload) {
    let ConstraintTypeEnum;
    (function (ConstraintTypeEnum) {
        ConstraintTypeEnum[ConstraintTypeEnum["Transaction"] = 'transaction'] = "Transaction";
        ConstraintTypeEnum[ConstraintTypeEnum["Amount"] = 'amount'] = "Amount";
    })(ConstraintTypeEnum = CreateBalanceLimitPayload.ConstraintTypeEnum || (CreateBalanceLimitPayload.ConstraintTypeEnum = {}));
    let DurationUnitEnum;
    (function (DurationUnitEnum) {
        DurationUnitEnum[DurationUnitEnum["Day"] = 'day'] = "Day";
        DurationUnitEnum[DurationUnitEnum["Week"] = 'week'] = "Week";
        DurationUnitEnum[DurationUnitEnum["Month"] = 'month'] = "Month";
        DurationUnitEnum[DurationUnitEnum["Year"] = 'year'] = "Year";
    })(DurationUnitEnum = CreateBalanceLimitPayload.DurationUnitEnum || (CreateBalanceLimitPayload.DurationUnitEnum = {}));
    let TransactionTypeEnum;
    (function (TransactionTypeEnum) {
        TransactionTypeEnum[TransactionTypeEnum["Credit"] = 'credit'] = "Credit";
        TransactionTypeEnum[TransactionTypeEnum["Debit"] = 'debit'] = "Debit";
    })(TransactionTypeEnum = CreateBalanceLimitPayload.TransactionTypeEnum || (CreateBalanceLimitPayload.TransactionTypeEnum = {}));
})(CreateBalanceLimitPayload || (exports.CreateBalanceLimitPayload = CreateBalanceLimitPayload = {}));
