// @getbrevo/brevo/dist/model/crmAttributesPostRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CrmAttributesPostRequest = void 0;
class CrmAttributesPostRequest {
    static getAttributeTypeMap() {
        return CrmAttributesPostRequest.attributeTypeMap;
    }
}
exports.CrmAttributesPostRequest = CrmAttributesPostRequest;
CrmAttributesPostRequest.discriminator = undefined;
CrmAttributesPostRequest.attributeTypeMap = [
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "attributeType",
        "baseName": "attributeType",
        "type": "CrmAttributesPostRequest.AttributeTypeEnum"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "optionsLabels",
        "baseName": "optionsLabels",
        "type": "Array<string>"
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "CrmAttributesPostRequest.ObjectTypeEnum"
    }
];
(function (CrmAttributesPostRequest) {
    let AttributeTypeEnum;
    (function (AttributeTypeEnum) {
        AttributeTypeEnum[AttributeTypeEnum["Text"] = 'text'] = "Text";
        AttributeTypeEnum[AttributeTypeEnum["User"] = 'user'] = "User";
        AttributeTypeEnum[AttributeTypeEnum["Number"] = 'number'] = "Number";
        AttributeTypeEnum[AttributeTypeEnum["SingleSelect"] = 'single-select'] = "SingleSelect";
        AttributeTypeEnum[AttributeTypeEnum["Date"] = 'date'] = "Date";
        AttributeTypeEnum[AttributeTypeEnum["Boolean"] = 'boolean'] = "Boolean";
        AttributeTypeEnum[AttributeTypeEnum["MultiChoice"] = 'multi-choice'] = "MultiChoice";
    })(AttributeTypeEnum = CrmAttributesPostRequest.AttributeTypeEnum || (CrmAttributesPostRequest.AttributeTypeEnum = {}));
    let ObjectTypeEnum;
    (function (ObjectTypeEnum) {
        ObjectTypeEnum[ObjectTypeEnum["Companies"] = 'companies'] = "Companies";
        ObjectTypeEnum[ObjectTypeEnum["Deals"] = 'deals'] = "Deals";
    })(ObjectTypeEnum = CrmAttributesPostRequest.ObjectTypeEnum || (CrmAttributesPostRequest.ObjectTypeEnum = {}));
})(CrmAttributesPostRequest || (exports.CrmAttributesPostRequest = CrmAttributesPostRequest = {}));
