// @getbrevo/brevo/dist/model/mainReward.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MainReward = void 0;
class MainReward {
    constructor() {
        this['expirationModifier'] = MainReward.ExpirationModifierEnum.NoModification;
    }
    static getAttributeTypeMap() {
        return MainReward.attributeTypeMap;
    }
}
exports.MainReward = MainReward;
MainReward.discriminator = undefined;
MainReward.attributeTypeMap = [
    {
        "name": "attributionPerConsumer",
        "baseName": "attributionPerConsumer",
        "type": "number"
    },
    {
        "name": "balanceDefinitionId",
        "baseName": "balanceDefinitionId",
        "type": "string"
    },
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "codeCount",
        "baseName": "codeCount",
        "type": "number"
    },
    {
        "name": "codeGeneratorId",
        "baseName": "codeGeneratorId",
        "type": "string"
    },
    {
        "name": "codePoolId",
        "baseName": "codePoolId",
        "type": "string"
    },
    {
        "name": "config",
        "baseName": "config",
        "type": "string"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "disabledAt",
        "baseName": "disabledAt",
        "type": "Date"
    },
    {
        "name": "endDate",
        "baseName": "endDate",
        "type": "Date"
    },
    {
        "name": "expirationDate",
        "baseName": "expirationDate",
        "type": "Date"
    },
    {
        "name": "expirationModifier",
        "baseName": "expirationModifier",
        "type": "MainReward.ExpirationModifierEnum"
    },
    {
        "name": "expirationUnit",
        "baseName": "expirationUnit",
        "type": "string"
    },
    {
        "name": "expirationValue",
        "baseName": "expirationValue",
        "type": "number"
    },
    {
        "name": "generator",
        "baseName": "generator",
        "type": "MainGenerator"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "limits",
        "baseName": "limits",
        "type": "Array<MainLimit>"
    },
    {
        "name": "loyaltyProgramId",
        "baseName": "loyaltyProgramId",
        "type": "string"
    },
    {
        "name": "meta",
        "baseName": "meta",
        "type": "{ [key: string]: object; }"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "products",
        "baseName": "products",
        "type": "Array<MainProduct>"
    },
    {
        "name": "publicDescription",
        "baseName": "publicDescription",
        "type": "string"
    },
    {
        "name": "publicImage",
        "baseName": "publicImage",
        "type": "string"
    },
    {
        "name": "publicName",
        "baseName": "publicName",
        "type": "string"
    },
    {
        "name": "redeemPerConsumer",
        "baseName": "redeemPerConsumer",
        "type": "number"
    },
    {
        "name": "redeemRules",
        "baseName": "redeemRules",
        "type": "Array<string>"
    },
    {
        "name": "rewardConfigs",
        "baseName": "rewardConfigs",
        "type": "MainRewardConfigurations"
    },
    {
        "name": "rule",
        "baseName": "rule",
        "type": "MainRule"
    },
    {
        "name": "startDate",
        "baseName": "startDate",
        "type": "Date"
    },
    {
        "name": "subtractBalanceDefinitionId",
        "baseName": "subtractBalanceDefinitionId",
        "type": "string"
    },
    {
        "name": "subtractBalanceStrategy",
        "baseName": "subtractBalanceStrategy",
        "type": "string"
    },
    {
        "name": "subtractBalanceValue",
        "baseName": "subtractBalanceValue",
        "type": "number"
    },
    {
        "name": "subtractTotalBalance",
        "baseName": "subtractTotalBalance",
        "type": "boolean"
    },
    {
        "name": "totalAttribution",
        "baseName": "totalAttribution",
        "type": "number"
    },
    {
        "name": "totalRedeem",
        "baseName": "totalRedeem",
        "type": "number"
    },
    {
        "name": "triggerId",
        "baseName": "triggerId",
        "type": "string"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "string"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "string"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    },
    {
        "name": "valueType",
        "baseName": "valueType",
        "type": "string"
    }
];
(function (MainReward) {
    let ExpirationModifierEnum;
    (function (ExpirationModifierEnum) {
        ExpirationModifierEnum[ExpirationModifierEnum["StartOfPeriod"] = 'startOfPeriod'] = "StartOfPeriod";
        ExpirationModifierEnum[ExpirationModifierEnum["EndOfPeriod"] = 'endOfPeriod'] = "EndOfPeriod";
        ExpirationModifierEnum[ExpirationModifierEnum["NoModification"] = 'noModification'] = "NoModification";
    })(ExpirationModifierEnum = MainReward.ExpirationModifierEnum || (MainReward.ExpirationModifierEnum = {}));
})(MainReward || (exports.MainReward = MainReward = {}));
