// @getbrevo/brevo/dist/model/mainGetContactRewardsPayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MainGetContactRewardsPayload = void 0;
class MainGetContactRewardsPayload {
    static getAttributeTypeMap() {
        return MainGetContactRewardsPayload.attributeTypeMap;
    }
}
exports.MainGetContactRewardsPayload = MainGetContactRewardsPayload;
MainGetContactRewardsPayload.discriminator = undefined;
MainGetContactRewardsPayload.attributeTypeMap = [
    {
        "name": "contactId",
        "baseName": "contactId",
        "type": "number"
    },
    {
        "name": "limit",
        "baseName": "limit",
        "type": "number"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "Array<MainFilter>"
    },
    {
        "name": "offset",
        "baseName": "offset",
        "type": "number"
    },
    {
        "name": "rewardId",
        "baseName": "rewardId",
        "type": "string"
    },
    {
        "name": "sort",
        "baseName": "sort",
        "type": "MainGetContactRewardsPayload.SortEnum"
    },
    {
        "name": "sortField",
        "baseName": "sortField",
        "type": "MainGetContactRewardsPayload.SortFieldEnum"
    }
];
(function (MainGetContactRewardsPayload) {
    let SortEnum;
    (function (SortEnum) {
        SortEnum[SortEnum["Asc"] = 'asc'] = "Asc";
        SortEnum[SortEnum["Desc"] = 'desc'] = "Desc";
    })(SortEnum = MainGetContactRewardsPayload.SortEnum || (MainGetContactRewardsPayload.SortEnum = {}));
    let SortFieldEnum;
    (function (SortFieldEnum) {
        SortFieldEnum[SortFieldEnum["UpdatedAt"] = 'updatedAt'] = "UpdatedAt";
        SortFieldEnum[SortFieldEnum["CreatedAt"] = 'createdAt'] = "CreatedAt";
    })(SortFieldEnum = MainGetContactRewardsPayload.SortFieldEnum || (MainGetContactRewardsPayload.SortFieldEnum = {}));
})(MainGetContactRewardsPayload || (exports.MainGetContactRewardsPayload = MainGetContactRewardsPayload = {}));
