// @getbrevo/brevo/dist/model/tierGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TierGroup = void 0;
class TierGroup {
    constructor() {
        this['upgradeStrategy'] = TierGroup.UpgradeStrategyEnum.RealTime;
        this['downgradeStrategy'] = TierGroup.DowngradeStrategyEnum.RealTime;
    }
    static getAttributeTypeMap() {
        return TierGroup.attributeTypeMap;
    }
}
exports.TierGroup = TierGroup;
TierGroup.discriminator = undefined;
TierGroup.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "tierOrder",
        "baseName": "tierOrder",
        "type": "Array<string>"
    },
    {
        "name": "loyaltyProgramId",
        "baseName": "loyaltyProgramId",
        "type": "string"
    },
    {
        "name": "upgradeStrategy",
        "baseName": "upgradeStrategy",
        "type": "TierGroup.UpgradeStrategyEnum"
    },
    {
        "name": "downgradeStrategy",
        "baseName": "downgradeStrategy",
        "type": "TierGroup.DowngradeStrategyEnum"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    }
];
(function (TierGroup) {
    let UpgradeStrategyEnum;
    (function (UpgradeStrategyEnum) {
        UpgradeStrategyEnum[UpgradeStrategyEnum["RealTime"] = 'real_time'] = "RealTime";
        UpgradeStrategyEnum[UpgradeStrategyEnum["MembershipAnniversary"] = 'membership_anniversary'] = "MembershipAnniversary";
        UpgradeStrategyEnum[UpgradeStrategyEnum["TierAnniversary"] = 'tier_anniversary'] = "TierAnniversary";
    })(UpgradeStrategyEnum = TierGroup.UpgradeStrategyEnum || (TierGroup.UpgradeStrategyEnum = {}));
    let DowngradeStrategyEnum;
    (function (DowngradeStrategyEnum) {
        DowngradeStrategyEnum[DowngradeStrategyEnum["RealTime"] = 'real_time'] = "RealTime";
        DowngradeStrategyEnum[DowngradeStrategyEnum["MembershipAnniversary"] = 'membership_anniversary'] = "MembershipAnniversary";
        DowngradeStrategyEnum[DowngradeStrategyEnum["TierAnniversary"] = 'tier_anniversary'] = "TierAnniversary";
    })(DowngradeStrategyEnum = TierGroup.DowngradeStrategyEnum || (TierGroup.DowngradeStrategyEnum = {}));
})(TierGroup || (exports.TierGroup = TierGroup = {}));
