// @getbrevo/brevo/dist/model/updateBalanceDefinitionPayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateBalanceDefinitionPayload = void 0;
class UpdateBalanceDefinitionPayload {
    static getAttributeTypeMap() {
        return UpdateBalanceDefinitionPayload.attributeTypeMap;
    }
}
exports.UpdateBalanceDefinitionPayload = UpdateBalanceDefinitionPayload;
UpdateBalanceDefinitionPayload.discriminator = undefined;
UpdateBalanceDefinitionPayload.attributeTypeMap = [
    {
        "name": "balanceAvailabilityDurationModifier",
        "baseName": "balanceAvailabilityDurationModifier",
        "type": "UpdateBalanceDefinitionPayload.BalanceAvailabilityDurationModifierEnum"
    },
    {
        "name": "balanceAvailabilityDurationUnit",
        "baseName": "balanceAvailabilityDurationUnit",
        "type": "UpdateBalanceDefinitionPayload.BalanceAvailabilityDurationUnitEnum"
    },
    {
        "name": "balanceAvailabilityDurationValue",
        "baseName": "balanceAvailabilityDurationValue",
        "type": "number"
    },
    {
        "name": "balanceExpirationDate",
        "baseName": "balanceExpirationDate",
        "type": "string"
    },
    {
        "name": "balanceOptionAmountOvertakingStrategy",
        "baseName": "balanceOptionAmountOvertakingStrategy",
        "type": "UpdateBalanceDefinitionPayload.BalanceOptionAmountOvertakingStrategyEnum"
    },
    {
        "name": "balanceOptionCreditRounding",
        "baseName": "balanceOptionCreditRounding",
        "type": "UpdateBalanceDefinitionPayload.BalanceOptionCreditRoundingEnum"
    },
    {
        "name": "balanceOptionDebitRounding",
        "baseName": "balanceOptionDebitRounding",
        "type": "UpdateBalanceDefinitionPayload.BalanceOptionDebitRoundingEnum"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "imageRef",
        "baseName": "imageRef",
        "type": "string"
    },
    {
        "name": "maxAmount",
        "baseName": "maxAmount",
        "type": "number"
    },
    {
        "name": "maxCreditAmountLimit",
        "baseName": "maxCreditAmountLimit",
        "type": "number"
    },
    {
        "name": "maxDebitAmountLimit",
        "baseName": "maxDebitAmountLimit",
        "type": "number"
    },
    {
        "name": "meta",
        "baseName": "meta",
        "type": "{ [key: string]: any; }"
    },
    {
        "name": "minAmount",
        "baseName": "minAmount",
        "type": "number"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "UpdateBalanceDefinitionPayload.UnitEnum"
    }
];
(function (UpdateBalanceDefinitionPayload) {
    let BalanceAvailabilityDurationModifierEnum;
    (function (BalanceAvailabilityDurationModifierEnum) {
        BalanceAvailabilityDurationModifierEnum[BalanceAvailabilityDurationModifierEnum["NoModification"] = 'noModification'] = "NoModification";
        BalanceAvailabilityDurationModifierEnum[BalanceAvailabilityDurationModifierEnum["StartOfPeriod"] = 'startOfPeriod'] = "StartOfPeriod";
        BalanceAvailabilityDurationModifierEnum[BalanceAvailabilityDurationModifierEnum["EndOfPeriod"] = 'endOfPeriod'] = "EndOfPeriod";
    })(BalanceAvailabilityDurationModifierEnum = UpdateBalanceDefinitionPayload.BalanceAvailabilityDurationModifierEnum || (UpdateBalanceDefinitionPayload.BalanceAvailabilityDurationModifierEnum = {}));
    let BalanceAvailabilityDurationUnitEnum;
    (function (BalanceAvailabilityDurationUnitEnum) {
        BalanceAvailabilityDurationUnitEnum[BalanceAvailabilityDurationUnitEnum["Day"] = 'day'] = "Day";
        BalanceAvailabilityDurationUnitEnum[BalanceAvailabilityDurationUnitEnum["Week"] = 'week'] = "Week";
        BalanceAvailabilityDurationUnitEnum[BalanceAvailabilityDurationUnitEnum["Month"] = 'month'] = "Month";
        BalanceAvailabilityDurationUnitEnum[BalanceAvailabilityDurationUnitEnum["Year"] = 'year'] = "Year";
    })(BalanceAvailabilityDurationUnitEnum = UpdateBalanceDefinitionPayload.BalanceAvailabilityDurationUnitEnum || (UpdateBalanceDefinitionPayload.BalanceAvailabilityDurationUnitEnum = {}));
    let BalanceOptionAmountOvertakingStrategyEnum;
    (function (BalanceOptionAmountOvertakingStrategyEnum) {
        BalanceOptionAmountOvertakingStrategyEnum[BalanceOptionAmountOvertakingStrategyEnum["Strict"] = 'strict'] = "Strict";
        BalanceOptionAmountOvertakingStrategyEnum[BalanceOptionAmountOvertakingStrategyEnum["Partial"] = 'partial'] = "Partial";
    })(BalanceOptionAmountOvertakingStrategyEnum = UpdateBalanceDefinitionPayload.BalanceOptionAmountOvertakingStrategyEnum || (UpdateBalanceDefinitionPayload.BalanceOptionAmountOvertakingStrategyEnum = {}));
    let BalanceOptionCreditRoundingEnum;
    (function (BalanceOptionCreditRoundingEnum) {
        BalanceOptionCreditRoundingEnum[BalanceOptionCreditRoundingEnum["Lower"] = 'lower'] = "Lower";
        BalanceOptionCreditRoundingEnum[BalanceOptionCreditRoundingEnum["Upper"] = 'upper'] = "Upper";
        BalanceOptionCreditRoundingEnum[BalanceOptionCreditRoundingEnum["Natural"] = 'natural'] = "Natural";
    })(BalanceOptionCreditRoundingEnum = UpdateBalanceDefinitionPayload.BalanceOptionCreditRoundingEnum || (UpdateBalanceDefinitionPayload.BalanceOptionCreditRoundingEnum = {}));
    let BalanceOptionDebitRoundingEnum;
    (function (BalanceOptionDebitRoundingEnum) {
        BalanceOptionDebitRoundingEnum[BalanceOptionDebitRoundingEnum["Lower"] = 'lower'] = "Lower";
        BalanceOptionDebitRoundingEnum[BalanceOptionDebitRoundingEnum["Upper"] = 'upper'] = "Upper";
        BalanceOptionDebitRoundingEnum[BalanceOptionDebitRoundingEnum["Natural"] = 'natural'] = "Natural";
    })(BalanceOptionDebitRoundingEnum = UpdateBalanceDefinitionPayload.BalanceOptionDebitRoundingEnum || (UpdateBalanceDefinitionPayload.BalanceOptionDebitRoundingEnum = {}));
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Points"] = 'POINTS'] = "Points";
        UnitEnum[UnitEnum["Eur"] = 'EUR'] = "Eur";
        UnitEnum[UnitEnum["Usd"] = 'USD'] = "Usd";
        UnitEnum[UnitEnum["Mxn"] = 'MXN'] = "Mxn";
        UnitEnum[UnitEnum["Gbp"] = 'GBP'] = "Gbp";
        UnitEnum[UnitEnum["Inr"] = 'INR'] = "Inr";
        UnitEnum[UnitEnum["Cad"] = 'CAD'] = "Cad";
        UnitEnum[UnitEnum["Sgd"] = 'SGD'] = "Sgd";
        UnitEnum[UnitEnum["Ron"] = 'RON'] = "Ron";
        UnitEnum[UnitEnum["Jpy"] = 'JPY'] = "Jpy";
        UnitEnum[UnitEnum["Myr"] = 'MYR'] = "Myr";
        UnitEnum[UnitEnum["Clp"] = 'CLP'] = "Clp";
        UnitEnum[UnitEnum["Pen"] = 'PEN'] = "Pen";
        UnitEnum[UnitEnum["Mad"] = 'MAD'] = "Mad";
        UnitEnum[UnitEnum["Aud"] = 'AUD'] = "Aud";
        UnitEnum[UnitEnum["Chf"] = 'CHF'] = "Chf";
        UnitEnum[UnitEnum["Brl"] = 'BRL'] = "Brl";
    })(UnitEnum = UpdateBalanceDefinitionPayload.UnitEnum || (UpdateBalanceDefinitionPayload.UnitEnum = {}));
})(UpdateBalanceDefinitionPayload || (exports.UpdateBalanceDefinitionPayload = UpdateBalanceDefinitionPayload = {}));
