// @getbrevo/brevo/dist/model/updateTierGroupRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTierGroupRequest = void 0;
class UpdateTierGroupRequest {
    constructor() {
        this['upgradeStrategy'] = UpdateTierGroupRequest.UpgradeStrategyEnum.RealTime;
        this['downgradeStrategy'] = UpdateTierGroupRequest.DowngradeStrategyEnum.RealTime;
    }
    static getAttributeTypeMap() {
        return UpdateTierGroupRequest.attributeTypeMap;
    }
}
exports.UpdateTierGroupRequest = UpdateTierGroupRequest;
UpdateTierGroupRequest.discriminator = undefined;
UpdateTierGroupRequest.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "tierOrder",
        "baseName": "tierOrder",
        "type": "Array<string>"
    },
    {
        "name": "upgradeStrategy",
        "baseName": "upgradeStrategy",
        "type": "UpdateTierGroupRequest.UpgradeStrategyEnum"
    },
    {
        "name": "downgradeStrategy",
        "baseName": "downgradeStrategy",
        "type": "UpdateTierGroupRequest.DowngradeStrategyEnum"
    }
];
(function (UpdateTierGroupRequest) {
    let UpgradeStrategyEnum;
    (function (UpgradeStrategyEnum) {
        UpgradeStrategyEnum[UpgradeStrategyEnum["RealTime"] = 'real_time'] = "RealTime";
        UpgradeStrategyEnum[UpgradeStrategyEnum["MembershipAnniversary"] = 'membership_anniversary'] = "MembershipAnniversary";
        UpgradeStrategyEnum[UpgradeStrategyEnum["TierAnniversary"] = 'tier_anniversary'] = "TierAnniversary";
    })(UpgradeStrategyEnum = UpdateTierGroupRequest.UpgradeStrategyEnum || (UpdateTierGroupRequest.UpgradeStrategyEnum = {}));
    let DowngradeStrategyEnum;
    (function (DowngradeStrategyEnum) {
        DowngradeStrategyEnum[DowngradeStrategyEnum["RealTime"] = 'real_time'] = "RealTime";
        DowngradeStrategyEnum[DowngradeStrategyEnum["MembershipAnniversary"] = 'membership_anniversary'] = "MembershipAnniversary";
        DowngradeStrategyEnum[DowngradeStrategyEnum["TierAnniversary"] = 'tier_anniversary'] = "TierAnniversary";
    })(DowngradeStrategyEnum = UpdateTierGroupRequest.DowngradeStrategyEnum || (UpdateTierGroupRequest.DowngradeStrategyEnum = {}));
})(UpdateTierGroupRequest || (exports.UpdateTierGroupRequest = UpdateTierGroupRequest = {}));
