// @getbrevo/brevo/dist/model/updateBalanceLimitPayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateBalanceLimitPayload = void 0;
class UpdateBalanceLimitPayload {
    static getAttributeTypeMap() {
        return UpdateBalanceLimitPayload.attributeTypeMap;
    }
}
exports.UpdateBalanceLimitPayload = UpdateBalanceLimitPayload;
UpdateBalanceLimitPayload.discriminator = undefined;
UpdateBalanceLimitPayload.attributeTypeMap = [
    {
        "name": "constraintType",
        "baseName": "constraintType",
        "type": "UpdateBalanceLimitPayload.ConstraintTypeEnum"
    },
    {
        "name": "durationUnit",
        "baseName": "durationUnit",
        "type": "UpdateBalanceLimitPayload.DurationUnitEnum"
    },
    {
        "name": "durationValue",
        "baseName": "durationValue",
        "type": "number"
    },
    {
        "name": "slidingSchedule",
        "baseName": "slidingSchedule",
        "type": "boolean"
    },
    {
        "name": "transactionType",
        "baseName": "transactionType",
        "type": "UpdateBalanceLimitPayload.TransactionTypeEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    }
];
(function (UpdateBalanceLimitPayload) {
    let ConstraintTypeEnum;
    (function (ConstraintTypeEnum) {
        ConstraintTypeEnum[ConstraintTypeEnum["Transaction"] = 'transaction'] = "Transaction";
        ConstraintTypeEnum[ConstraintTypeEnum["Amount"] = 'amount'] = "Amount";
    })(ConstraintTypeEnum = UpdateBalanceLimitPayload.ConstraintTypeEnum || (UpdateBalanceLimitPayload.ConstraintTypeEnum = {}));
    let DurationUnitEnum;
    (function (DurationUnitEnum) {
        DurationUnitEnum[DurationUnitEnum["Day"] = 'day'] = "Day";
        DurationUnitEnum[DurationUnitEnum["Week"] = 'week'] = "Week";
        DurationUnitEnum[DurationUnitEnum["Month"] = 'month'] = "Month";
        DurationUnitEnum[DurationUnitEnum["Year"] = 'year'] = "Year";
    })(DurationUnitEnum = UpdateBalanceLimitPayload.DurationUnitEnum || (UpdateBalanceLimitPayload.DurationUnitEnum = {}));
    let TransactionTypeEnum;
    (function (TransactionTypeEnum) {
        TransactionTypeEnum[TransactionTypeEnum["Credit"] = 'credit'] = "Credit";
        TransactionTypeEnum[TransactionTypeEnum["Debit"] = 'debit'] = "Debit";
    })(TransactionTypeEnum = UpdateBalanceLimitPayload.TransactionTypeEnum || (UpdateBalanceLimitPayload.TransactionTypeEnum = {}));
})(UpdateBalanceLimitPayload || (exports.UpdateBalanceLimitPayload = UpdateBalanceLimitPayload = {}));
